packages/core/theme/src/utils/object.ts

```diff
@@ -52,5 +49,5 @@ export const flattenThemeObject = <TTarget>(obj: TTarget) =>
     flatten(obj, {
       safe: true,
       delimiter: "-",
-    }) as Object,
+    }) as StringMap,
```

Comment:
the value could be boolean. See `safe`.

Reply:
From the existing string type, we modified it by creating a ColorMap type including a boolean type.